third_party/README.md

```diff
@@ -0,0 +1,11 @@
+## memcached_ibuf
```

Comment:
I would place this text to another place (`README.md` in a root directory?) or at least add a link to this README to the main `README.me`.

Reply:
It is for developers and even especially for developers who want to understand why we don't just use `ibuf.[ch]` from small and when we can drop the copy. I would not mix it with user documentation.

I had thought that if someone will look at `third_party/memcached_ibuf.[ch]`, (s)he will find `third_party/README.md`.

OTOH, we can collect all arcane knowledge in `HACKING.md` in the repository root and link to it from `README.md`.

Reply:
Moved to `HACKING.md`, added the link from `README.md`.